hessian/src/main/java/com/linecorp/armeria/internal/client/hessian/HessianHttpClientDelegate.java

```diff
@@ -0,0 +1,369 @@
+
+            try {
+                try (ByteBufOutputStream os = new ByteBufOutputStream(buf)) {
+                    final AbstractHessianOutput out = factory.getHessianOutput(os, options);
+                    out.call(method, args.toArray());
+                    out.flush();
+                }
+                ctx.logBuilder().requestContent(call, null);
+            } catch (Throwable t) {
+                buf.release();
+                throw castException(t);
+            }
+
+            @Nullable
+            final Endpoint endpoint = ctx.endpoint();
+
+            final HttpRequest httpReq = HttpRequest.of(
+                    RequestHeaders.builder(HttpMethod.POST, ctx.path())
+                                  .scheme(ctx.sessionProtocol()).authority(
+                                          endpoint != null ? endpoint.authority() : "UNKNOWN")
+                                  .contentType(mediaType)
+                                  // X-Hessian-* not standard header.
+                                  .set("X-Hessian-Service", call.serviceName())
+                                  .set("X-Hessian-Method", call.method())
+                                  .build(),
+                    HttpData.wrap(buf).withEndOfStream());
+
+            ctx.updateRequest(httpReq);
+            ctx.logBuilder().defer(RequestLogProperty.RESPONSE_CONTENT);
+
+            final HttpResponse httpResponse;
+            try {
+                httpResponse = unwrap().execute(ctx, httpReq);
+            } catch (Throwable t) {
+                httpReq.abort();
+                throw castException(t);
+            }
+
+            httpResponse.aggregateWithPooledObjects(ctx.eventLoop(), ctx.alloc()).handle((res, cause) -> {
+                if (cause != null) {
+                    handlePreDecodeException(ctx, reply, Exceptions.peel(cause));
+                    return null;
+                }
+
+                try (HttpData content = res.content()) {
+                    final HttpStatus status = res.status();
+                    if (status.code() != HttpStatus.OK.code()) {
+                        handlePreDecodeException(ctx, reply,
+                                                 new InvalidResponseHeadersException(res.headers()));
+                        return null;
+                    }
+
+                    try {
+                        handle(ctx, reply, func, content);
+                    } catch (Throwable t) {
+                        handlePreDecodeException(ctx, reply, t);
+                    }
+                }
+
+                return null;
+            }).exceptionally(CompletionActions::log);
+        } catch (Throwable cause) {
+            handlePreDecodeException(ctx, reply, cause);
+        }
+
+        return reply;
+    }
+
+    private HessianServiceClientMetadata metadata(Class<?> serviceType) {
+        final HessianServiceClientMetadata metadata = metadataMap.get(serviceType);
```

Comment:
Is the `serviceType` determined when `HessianHttpClientDelegate` is created?
If so, could we create a `HessianServiceClientMetadata` when this class is initiated?